### PR TITLE
Handle pkg_info(1) error message "Can't find"

### PR DESCRIPTION
**Pull request by sizeofvoid**
##### SUMMARY
**OpenBSD pkg_info(1) behavior change.**

As we can see pkg_info behaves different between OpenBSD -current (>7.3) and OpenBSD <= 7.3.
It prints "Can't find inst:PKGNAME" if the package is not present. This has to be handled in the module otherwise we see

```
failed: [dev] (item=ranger) => {"ansible_loop_var": "item", "changed": false, "item": "ranger", "msg": "failed in get_package_state(): Can't find inst:ranger\n"}
```
 
**Example pkg_info(1) call on OpenBSD 7.3 -stable**
```
$ dmesg | head
OpenBSD 7.3 (RAMDISK_CD) #1063: Sat Mar 25 10:41:49 MDT 2023
    deraadt@amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/RAMDISK_CD
$ pkg_info -Iq inst:abook
abook-0.6.1p3
triangle$ echo $?
0
$ pkg_info -Iq inst:NOTINSTALLED
$ echo $?
1
```
**Example pkg_info(1) call on OpenBSD -current**
```
$ dmesg | head
OpenBSD 7.3-current (GENERIC.MP) #1247: Sun Jun 18 09:42:58 MDT 2023
    deraadt@amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/GENERIC.MP
$ pkg_info -Iq inst:abook
abook-0.6.1p3
$ echo $?
0
$ pkg_info -Iq inst:NOTINSTALLED
Can't find inst:NOTINSTALLED
$ echo $?
1
```

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
openbsd_pkg

### CODE REVIEW

File: plugins/modules/openbsd_pkg.py [hunk 1]
```diff
@@ -169,7 +169,11 @@ def get_package_state(names, pkg_spec, module):
         rc, stdout, stderr = execute_command(command, module)
 
         if stderr:
-            module.fail_json(msg="failed in get_package_state(): " + stderr)
+            match = "Can\'t find inst:%s" % (name) in stderr
```
Comment (felixfontein) [inline plugins/modules/openbsd_pkg.py:172]: I think this way it's easier to understand what it does:
```suggestion
            match = ("Can't find inst:%s" % name) in stderr
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: changelogs/fragments/6785-openbsd_pkg_pkg_info_handling.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+xes:
```
Comment (felixfontein) [inline changelogs/fragments/6785-openbsd_pkg_pkg_info_handling.yml:1]:
```suggestion
bugfixes:
```

Comment (sizeofvoid) [reply changelogs/fragments/6785-openbsd_pkg_pkg_info_handling.yml]: Thanks, no commit before the first coffee ;)

File: changelogs/fragments/6785-openbsd_pkg_pkg_info_handling.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - openbsd_pkg - The pkg_info(1) behavior has changed in OpenBSD >7.3. The error message ``Can't find`` should not lead to an error case (https://github.com/ansible-collections/community.general/pull/6785).
```
Comment (felixfontein) [inline changelogs/fragments/6785-openbsd_pkg_pkg_info_handling.yml:2]: Sorry, I also missed a detail while reviewing (damn early mornings ;) )
```suggestion
  - openbsd_pkg - the pkg_info(1) behavior has changed in OpenBSD >7.3. The error message ``Can't find`` should not lead to an error case (https://github.com/ansible-collections/community.general/pull/6785).
```

**Comment by felixfontein** [comment]
Looks good to me, as far as I can tell. Will merge by the end of this week if nobody objects.

**Comment by eest** [comment]
Hello @sizeofvoid, 

Thanks for a solid bug report and patch, this looks good to me :). The only thing I am thinking about is that in most cases in the module we use fairly strict regexes for matching output as in `^this is the complete output$` to not let future changes in output mistakenly pass the test where this `("Can't find inst:%s" % name)` means it would also match `I still Can't find inst:NOTINSTALLED what kind of case is this?`.

Do you think it would be a good idea to go for a stricter regex match for this as well? I personally have preferred to be as strict as possible and have things fail rather than mistakenly work as things evolve, if that makes sense.

**Comment by sizeofvoid** [comment]
> Hello @sizeofvoid,
> 
> Do you think it would be a good idea to go for a stricter regex match for this as well? I personally have preferred to be as strict as possible and have things fail rather than mistakenly work as things evolve, if that makes sense.

Thanks @eest, I like the idea. I'll chance the current pull-request.

**Comment by eest** [comment]
This looks in line with what I had in mind, thanks! One more thing: since we are using the `name` variable as part of the regex we probably want to pass it through `re.escape()`. We have had problems in the past where package names include valid regex characters, see https://github.com/ansible-collections/community.general/commit/56b5be0630e226f20cf461b2ab9c722de5e34483

File: plugins/modules/openbsd_pkg.py [hunk 4]
```diff
@@ -169,7 +169,11 @@ def get_package_state(names, pkg_spec, module):
         rc, stdout, stderr = execute_command(command, module)
 
         if stderr:
-            module.fail_json(msg="failed in get_package_state(): " + stderr)
+            match = re.search(r"^Can't find inst:%s$" % name, stderr)
```
Comment (felixfontein) [inline plugins/modules/openbsd_pkg.py:172]:
```suggestion
            match = re.search(r"^Can't find inst:%s$" % re.escape(name), stderr)
```

**Review by russoz — APPROVED** [review]
LGTM

**Comment by eest** [comment]
Thanks for your work on this @sizeofvoid! Also thanks to @felixfontein for the ever present support work for things like writing changelog fragments etc :)

LGTM

**Comment by felixfontein** [comment]
@sizeofvoid thanks for your contribution!
@eest thanks for reviewing and the kind words!
@russoz thanks for reviewing as well!

**Pull request merged**